live-elo/src/exporter/websocket.rs

```diff
@@ -20,22 +20,42 @@ use super::shared_processor::SharedHandle;
 
 #[derive(Debug, Clone, PartialEq, Serialize)]
 #[serde(transparent)]
-pub struct PerformancePoints(f32);
+pub struct PerformancePoints(Option<f32>);
 
 // TODO: make this into a newtype smhsmh
 pub type LeaderboardPerformances = HashMap<AuthorId, PerformancePoints>;
 
 impl PerformancePoints {
     pub fn new(value: f32) -> Self {
-        Self(value)
+        Self(Some(value))
     }
 
-    pub fn get(&self) -> f32 {
+    pub fn get(&self) -> Option<f32> {
         self.0
     }
 
     pub const fn zero() -> Self {
-        Self(0.0)
+        Self(Some(0.0))
+    }
+
+    pub const fn none() -> Self {
+        Self(None)
+    }
+}
+
+impl ops::Add<&PerformancePoints> for &PerformancePoints {
+    type Output = PerformancePoints;
+
+    fn add(self, rhs: &PerformancePoints) -> Self::Output {
+        let lhs_val = self.get();
+        let rhs_val = rhs.get();
+
+        PerformancePoints::new(match (lhs_val, rhs_val) {
+            (None, None) => return PerformancePoints::none(),
+            (None, Some(x)) => x,
+            (Some(x), None) => x,
+            (Some(x), Some(y)) => x + y,
+        })
     }
```

Comment:
I think this looks a bit cleaner, but either works
```suggestion
    fn add(self, rhs: &PerformancePoints) -> Self::Output {
        match (self.get(), rhs.get()) {
            (None, None) => PerformancePoints::none(),
            (None, Some(x)) | (Some(x), None) => PerformancePoints::new(x),
            (Some(x), Some(y)) => PerformancePoints::new(x + y),
        }
    }
```